### PR TITLE
Review runtime checks

### LINKED ISSUE #422: Review Machine runtime checks
When we have all the assertions included, we should review the Machine to remove unnecessary runtime checks:
https://github.com/dylibso/chicory/pull/421#discussion_r1661270231

### PR DESCRIPTION
**Pull request by andreaTP**
Fix #422

### CODE REVIEW

File: runtime/src/main/java/com/dylibso/chicory/runtime/InterpreterMachine.java [hunk 1]
```diff
@@ -1696,14 +1682,12 @@ private static void I64_LOAD32_U(MStack stack, Instance instance, Operands opera
     private static void I64_LOAD32_S(MStack stack, Instance instance, Operands operands) {
         var ptr = readMemPtr(stack, operands);
         var val = instance.memory().readI32(ptr);
-        // TODO this is a bit hacky
```
Comment (evacchi) [inline runtime/src/main/java/com/dylibso/chicory/runtime/InterpreterMachine.java:1699]: curious: what was this about? 😬

File: runtime/src/main/java/com/dylibso/chicory/runtime/InterpreterMachine.java [hunk 2]
```diff
@@ -1630,11 +1620,7 @@ private static void F32_NEG(MStack stack) {
         stack.push(Value.floatToLong(-tos));
     }
 
-    private static void MEMORY_FILL(MStack stack, Instance instance, Operands operands) {
-        var memidx = (int) operands.get(0);
-        if (memidx != 0) {
-            throw new WASMRuntimeException("We don't support multiple memories just yet");
-        }
```
Comment (evacchi) [inline runtime/src/main/java/com/dylibso/chicory/runtime/InterpreterMachine.java:1637]: is this handled somewhere else now? e.g. validation?

File: runtime/src/main/java/com/dylibso/chicory/runtime/InterpreterMachine.java (hunk 1 above)
Comment (andreaTP) [reply runtime/src/main/java/com/dylibso/chicory/runtime/InterpreterMachine.java:1699]: I don't remember :shrug: an early day comment that got copy pasted around ...

File: runtime/src/main/java/com/dylibso/chicory/runtime/InterpreterMachine.java (hunk 2 above)
Comment (andreaTP) [reply runtime/src/main/java/com/dylibso/chicory/runtime/InterpreterMachine.java:1637]: yup: https://github.com/dylibso/chicory/blob/87ecca5f6d40fc15668df16885a5ddbf70b85497/wasm/src/main/java/com/dylibso/chicory/wasm/Validator.java#L583-L593

**Review by evacchi — APPROVED** [review]

**Pull request merged**